Incorrect usage is corrected for global variable

TimothyAsirJeyasing opened the pull request:
Incorrect usage of global variable is corrected
Spec file changes for python3 conversion

TimothyAsirJeyasing committed:
Update to python3 support

Python2 community support is EOL.
Changes done for python3 support.
Now tendrl-common will no longer support python2

Signed-off-by: Timothy Asir Jeyasingh <tjeyasin@redhat.com>

TimothyAsirJeyasing committed:
Build script changes for python3 conversion and bugfix

Global variable was used before it was assigned
Build script changes for python3 conversion.

Signed-off-by: Timothy Asir Jeyasingh <tjeyasin@redhat.com>

TimothyAsirJeyasing commented:
@amohan@redhat.com, @nthomas-redhat, @anmolbabu , @cloudbehl  Please review

cloudbehl commented:
@TimothyAsirJeyasing can you fix the `Codacy/PR Quality Review` and also why CI tests are failing?

cloudbehl commented on tendrl/commons/tests/flows/expand_cluster/test_ec_init.py:
```diff
@@ -1,4 +1,4 @@
-import __builtin__
+import builtins
```
use builtins instead of __builtin__ in code also. Check line 50

TimothyAsirJeyasing replied:
Done

cloudbehl reviewed (changes requested):
Please make this change in all the test files.

TimothyAsirJeyasing committed:
The __builtin__ module was renamed to builtins for python3 support

Signed-off-by: Timothy Asir Jeyasingh <tjeyasin@redhat.com>

aruniiird commented on setup.py:
```diff
@@ -7,7 +7,7 @@
 try:
     # Python 2 backwards compat
-    from __builtin__ import raw_input as input
+    from builtins import raw_input as input
```
the above line is giving following error in my python3 setup,
```python
>>> from builtins import raw_input as input
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'raw_input' from 'builtins' (unknown location)
>>> 
```
you may have to change it to,
```python
from six.moves import input
```
this will work in both python 2.7 and 3.x (tested on python 2.7.16 and python 3.7.5)

GowthamShanmugam replied:
can you try this: 
    from builtins import input

no need alias, raw_input is renamed as input

TimothyAsirJeyasing replied:
Thank you, I will do it!

TimothyAsirJeyasing commented:
@GowthamShanmugam , @shtripat Please review

aruniiird commented on tendrl/commons/utils/central_store/utils.py:
```diff
@@ -2,7 +2,7 @@
 import etcd
 import sys
-import thread
+import _thread
```
```suggestion
import _thread as thread
```
then you don't have to change anywhere else

TimothyAsirJeyasing replied:
Sure thank you

GowthamShanmugam commented on tendrl/commons/utils/central_store/fields.py:
```diff
@@ -244,7 +244,7 @@ def _set_value(self, value):
         # Force casting if we were given a caster.
         if self._caster:
-            for x in value.keys():
+            for x in list(value.keys()):
```
I am seeing these changes in a few places, it is not necessary to convert this list by default it will give the result in list only. 

But not sure please check :)

GowthamShanmugam replied:
no need list(value.keys()) or list((value.values()) or len(list((value.keys())) no need list conversion if i am right

TimothyAsirJeyasing replied:
Done

GowthamShanmugam reviewed (commented):
only minor suggestion, not change LGTM

Commit:
Fix missing methods in _thread and builtins

1. Update to use input function from builtins module
2. Remove unwanted list calls
3. Fix missing module issue for _thread

Signed-off-by: root <root@localhost.localdomain>

TimothyAsirJeyasing commented:
Have done the review changes. Please review @GowthamShanmugam , @aruniiird

Commit:
Remove unused variables

Signed-off-by: root <root@localhost.localdomain>

Commit:
Removed unused variables and enhance the code

Signed-off-by: Timothy Asir Jeyasingh <tjeyasin@redhat.com>

aruniiird commented on tendrl/commons/objects/cluster/atoms/import_cluster/__init__.py:
```diff
@@ -42,7 +42,7 @@ def run(self):
                 out, err, rc = cmd.run()
                 if not err:
-                    volumes = filter(None, out.split("\n"))
+                    volumes = [_f for _f in out.split("\n") if _f]
```
Instead of "\n" , isn't it a good idea to use, `os.linesep` (which will give more interoperability), but leaving this at your discretion.

TimothyAsirJeyasing replied:
This patch is mainly for python3 convert. However we can do this kind of enhancement at later point of time. Thank you.

aruniiird reviewed (commented):
A minor comment, nothing much to review here...

TimothyAsirJeyasing committed:
Fix indentation error and used os.linesep for spliting lines

Signed-off-by: Timothy Asir Jeyasingh <tjeyasin@redhat.com>
